values/encoding/int_encode.go

```diff
@@ -96,7 +96,7 @@ func (enc *intEncoder) Encode(
 	}
 
 	if len(dictionary) <= maxCardinalityAllowed {
-		bytesPerDictionaryValue := int64(math.Ceil(float64(bits.Len(uint(valueMeta.Max-valueMeta.Min))) / 8.0))
+		bytesPerDictionaryValue := int64(math.Max(math.Ceil(float64(bits.Len(uint(valueMeta.Max-valueMeta.Min)))/8.0), 1.0))
```

Comment:
Would you mind adding a few tests to make sure encoding/decoding/roundtrip an empty int/string value set works as expected? Thanks!